pysyun/conversation/workers/scheduler.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import pickle
+from datetime import datetime, timedelta
+
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from telegram.ext import Application
+
+from logger import logger_instance
+
+
+class Scheduler:
+    def __init__(self, minutes, hours, days, interval, message=None, state_transition=None):
+        self.state_machine = None
+        self.persistence_file = None
+        self.application = None
+        self.scheduler = AsyncIOScheduler()
+
+        self.minutes = minutes + hours * 60 + days * 24 * 3600
+        self.interval = interval
+
+        self.message = message
+        self.state_transition = state_transition
+
+    def set_application(self, application: Application):
+        self.application = application
+
+    def set_persistence_file(self, persistence_file):
+        self.persistence_file = persistence_file
+
+    def set_state_machine(self, state_machine):
+        self.state_machine = state_machine
+
+    def start(self):
+        logger_instance.info('Starting scheduler...')
+        if self.persistence_file and self.application:
+            if (self.state_transition and self.state_machine) or self.message:
+                self.scheduler.add_job(self.task, 'interval', minutes=self.interval)
+                self.scheduler.start()
+            else:
+                logger_instance.critical(f"Value error: either message or "
+                                         f"state_transition not specified")
+                raise ValueError("Message text or state_transition not specified")
```

Comment:
This should break on initializing the object. The object should not be possible to be initialized into a failing stage.